base/linalg/dense.jl

```diff
@@ -29,6 +33,41 @@ function scale!(X::Array{T}, s::Real) where T<:BlasComplex
     X
 end
 
+
+function isone(x::StridedMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    if m < div(ISONE_CUTOFF * m * m, sizeof(x))
+        _isone_triacheck(x, m)
+    else
+        _isone_fastcache(x, m)
```

Comment:
I'm not convinced by this name: I thought initially it meant that this is the version when the matrix fits in cache!